tuf/utils/x509.go

```diff
@@ -431,75 +450,53 @@ func ED25519ToPrivateKey(privKeyBytes []byte) (data.PrivateKey, error) {
 	return data.NewED25519PrivateKey(*pubKey, privKeyBytes)
 }
 
-func blockType(k data.PrivateKey) (string, error) {
-	switch k.Algorithm() {
-	case data.RSAKey, data.RSAx509Key:
-		return "RSA PRIVATE KEY", nil
-	case data.ECDSAKey, data.ECDSAx509Key:
-		return "EC PRIVATE KEY", nil
-	case data.ED25519Key:
-		return "ED25519 PRIVATE KEY", nil
-	default:
-		return "", fmt.Errorf("algorithm %s not supported", k.Algorithm())
-	}
-}
-
-// KeyToPEM returns a PEM encoded key from a Private Key
-func KeyToPEM(privKey data.PrivateKey, role data.RoleName, gun data.GUN) ([]byte, error) {
-	bt, err := blockType(privKey)
-	if err != nil {
-		return nil, err
-	}
-
-	headers := map[string]string{}
-	if role != "" {
-		headers["role"] = role.String()
-	}
-	if gun != "" {
-		headers["gun"] = gun.String()
+// ExtractPrivateKeyAttributes extracts role and gun values from private key bytes
+func ExtractPrivateKeyAttributes(pemBytes []byte) (data.RoleName, data.GUN, error) {
+	block, _ := pem.Decode(pemBytes)
+	if block == nil {
+		return "", "", errors.New("PEM block is empty")
 	}
 
-	block := &pem.Block{
-		Type:    bt,
-		Headers: headers,
-		Bytes:   privKey.Private(),
+	switch block.Type {
+	case "RSA PRIVATE KEY", "EC PRIVATE KEY", "ED25519 PRIVATE KEY":
+		if notary.FIPSEnabled() {
+			return "", "", errors.New("invalid key format")
```

Comment:
A similar error message to above would be better here: "<block type> not supported in FIPS mode"